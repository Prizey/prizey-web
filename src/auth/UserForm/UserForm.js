import React from 'react'
import isEmpty from 'lodash/isEmpty'
import { withStyles } from '@material-ui/core/styles'
import CircularProgress from '@material-ui/core/CircularProgress'
import { Formik, Form, Field } from 'formik'

import InputField from 'design/Field'
import ErrorComponent from 'design/Error/Error'

export const fieldList = [
  { label: 'Full Name', name: 'fullname' },
  { label: 'Address', name: 'address' },
  { label: 'City', name: 'city' },
  { label: 'State / Province / Region', name: 'stateProvinceRegion' },
  { label: 'Postal Code / ZIP', name: 'zipcode' },
  { label: 'Clothing Size', name: 'clothingSize' },
  { label: 'Shoe Size', name: 'shoeSize' },
]

const styles = theme => ({
  root: {
    marginTop: theme.spacing.sm,
  },
})

export default withStyles(styles)(props => (
  <Formik initialValues={props.user} onSubmit={props.onSubmit}>
    {() => (
      <Form className={props.classes.root}>
        {fieldList.map(field => (
          <Field
            key={field.name}
            component={InputField}
            name={field.name}
            label={field.label}
            type="text"
            validate={value => isEmpty(value) && 'required field'}
          />
        ))}

        <div style={{ textAlign: 'center' }}>
          {props.submitting && <CircularProgress color="primary" size={36} />}
          {props.submitError && (
            <ErrorComponent>{props.submitError}</ErrorComponent>
          )}

          {props.error && <ErrorComponent>{props.error}</ErrorComponent>}
        </div>

        {props.renderButton && props.renderButton(props)}
      </Form>
    )}
  </Formik>
))
