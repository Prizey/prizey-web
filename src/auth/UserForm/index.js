import Form from './UserForm'

export { fieldList } from './UserForm'
export default Form
