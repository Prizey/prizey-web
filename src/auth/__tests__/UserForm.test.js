import React from 'react'
import renderer from 'react-test-renderer'
import UserForm from '../UserForm'

jest.mock('formik', () => ({
  Field: props => <div {...props}>Field - {props.validate('')}</div>,
  Form: ({ children, ...props }) => <div {...props}>Form - {children}</div>,
  Formik: props => (
    <div {...props}>Formik - {props.render({ foo: 'bar' })}</div>
  ),
}))

it('renders correctly', () => {
  const params = {
    onSubmit: jest.fn(),
    renderButton: () => <button>Button</button>,
  }
  const tree = renderer.create(<UserForm {...params} />).toJSON()
  expect(tree).toMatchSnapshot()
})
