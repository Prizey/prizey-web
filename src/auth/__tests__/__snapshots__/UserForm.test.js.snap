// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  onSubmit={[MockFunction]}
>
  Formik - 
  <div
    className="Component-root-1"
  >
    Form - 
    <div
      component={[Function]}
      label="Full Name"
      name="fullname"
      type="text"
      validate={[Function]}
    >
      Field - 
      required field
    </div>
    <div
      component={[Function]}
      label="Address"
      name="address"
      type="text"
      validate={[Function]}
    >
      Field - 
      required field
    </div>
    <div
      component={[Function]}
      label="City"
      name="city"
      type="text"
      validate={[Function]}
    >
      Field - 
      required field
    </div>
    <div
      component={[Function]}
      label="State / Province / Region"
      name="stateProvinceRegion"
      type="text"
      validate={[Function]}
    >
      Field - 
      required field
    </div>
    <div
      component={[Function]}
      label="Postal Code / ZIP"
      name="zipcode"
      type="text"
      validate={[Function]}
    >
      Field - 
      required field
    </div>
    <div
      component={[Function]}
      label="Clothing Size"
      name="clothingSize"
      type="text"
      validate={[Function]}
    >
      Field - 
      required field
    </div>
    <div
      component={[Function]}
      label="Shoe Size"
      name="shoeSize"
      type="text"
      validate={[Function]}
    >
      Field - 
      required field
    </div>
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    />
    <button>
      Button
    </button>
  </div>
</div>
`;
